### PR TITLE
feat: added support for markdown to link an image

### LINKED ISSUE #1064: [FR] Support markdown to link an image
Linking Images
To add a link to an image, enclose the Markdown for the image in brackets, and then add the link in parentheses.

profile_page_appflowy_hactoberfest

Reference: 
* https://github.com/AppFlowy-IO/AppFlowy/discussions/1100
* https://www.markdownguide.org/basic-syntax/

### PR DESCRIPTION
**Pull request by jedau**
Closes #1064

Renders image using markdown. Tested on and verified as working in the examples plugin. This PR includes:
- Added handling for image markdown
- Combined image and link handlers (`markdownLinkToLinkHandler`) because the `ShortcutEvent` for the two scenarios are similar
- Improved RegEx for markdown links

Let me know if anyone has any questions, comments, or suggestions. I would be happy to address them all. Thanks!

### CODE REVIEW

File: frontend/app_flowy/packages/appflowy_editor/example/lib/main.dart [hunk 1]
```diff
@@ -123,6 +124,7 @@ class _MyHomePageState extends State<MyHomePage> {
                 'text/code_block': CodeBlockNodeWidgetBuilder(),
                 'tex': TeXBlockNodeWidgetBuidler(),
                 'horizontal_rule': HorizontalRuleWidgetBuilder(),
+                'network_image': NetworkImageNodeWidgetBuilder(),
```
Comment (LucasXu0) [inline frontend/app_flowy/packages/appflowy_editor/example/lib/main.dart:127]: I think it's not necessary to add this code. We're using `ImageNodeBuilder` to render the image node now.

File: frontend/app_flowy/packages/appflowy_editor/lib/src/service/internal_key_event_handlers/markdown_syntax_to_styled_text.dart [hunk 2]
```diff
@@ -188,57 +188,80 @@ ShortcutEventHandler doubleTildeToStrikethrough = (editorState, event) {
...
-      ),
-    );
-  editorState.commit();
+    // Delete the text and replace it with the image pointed to by the URL
+    editorState.transaction
+      ..deleteText(textNode, firstExclamation, text.length)
+      ..insertNode(
+          textNode.path,
+          Node.fromJson({
+            'type': 'network_image',
```
Comment (LucasXu0) [inline frontend/app_flowy/packages/appflowy_editor/lib/src/service/internal_key_event_handlers/markdown_syntax_to_styled_text.dart:227]: The `network_image` is deprecated. Please use the `image` type.

Here is an example.

```
{
  "type": "image",
  "attributes": {
    "image_src": "https://s1.ax1x.com/2022/08/26/v2sSbR.jpg",
    "align": "center"
  }
}
```

File: frontend/app_flowy/packages/appflowy_editor/lib/src/service/shortcut_event/built_in_shortcut_events.dart [hunk 3]
```diff
@@ -274,10 +273,10 @@ List<ShortcutEvent> builtInShortcutEvents = [
     handler: doubleTildeToStrikethrough,
   ),
   ShortcutEvent(
-    key: 'Markdown link to link',
-    command: 'shift+parenthesis right',
-    handler: markdownLinkToLinkHandler,
-  ),
+    key: 'Markdown link or image',
+    command: 'shift+digit 0',
```
Comment (LucasXu0) [inline frontend/app_flowy/packages/appflowy_editor/lib/src/service/shortcut_event/built_in_shortcut_events.dart:277]: No need to change the command. 

`)` = `shift+parentthesis right`.

Comment (jedau) [reply frontend/app_flowy/packages/appflowy_editor/lib/src/service/shortcut_event/built_in_shortcut_events.dart]: Addressed in [3963bd3](https://github.com/AppFlowy-IO/AppFlowy/pull/1277/commits/3963bd3e99bbac2460c31c970ecc8acd88acc486)

File: frontend/app_flowy/packages/appflowy_editor/lib/src/service/internal_key_event_handlers/markdown_syntax_to_styled_text.dart (hunk 2 above)
Comment (jedau) [reply frontend/app_flowy/packages/appflowy_editor/lib/src/service/internal_key_event_handlers/markdown_syntax_to_styled_text.dart]: Addressed in [3963bd3](https://github.com/AppFlowy-IO/AppFlowy/pull/1277/commits/3963bd3e99bbac2460c31c970ecc8acd88acc486)

File: frontend/app_flowy/packages/appflowy_editor/example/lib/main.dart (hunk 1 above)
Comment (jedau) [reply frontend/app_flowy/packages/appflowy_editor/example/lib/main.dart]: Addressed in [3963bd3](https://github.com/AppFlowy-IO/AppFlowy/pull/1277/commits/3963bd3e99bbac2460c31c970ecc8acd88acc486)

**Comment by LucasXu0** [comment]
LGTM. Thanks for contributing.

**Comment by jedau** [comment]
Thanks for the help getting it merged @LucasXu0 !

**Pull request merged**